test/kbaseauthcontroller/controller.go

```diff
@@ -119,50 +125,83 @@ func waitForStartup(port string) error {
 	return startupErr
 }
 
-func getClassPath(jarsDir string) (string, error) {
-	jarsDir, err := filepath.Abs(jarsDir)
+func getClassPath(auth2Jar string) (string, error) {
+	jpath, err := filepath.Abs(auth2Jar)
 	if err != nil {
 		return "", err
 	}
-	cp := []string(nil)
-	for _, j := range jars { // global variable, yech
-		jpath := path.Join(jarsDir, j)
-		if _, err := os.Stat(jpath); os.IsNotExist(err) {
-			return "", fmt.Errorf("Jar %v does not exist", jpath)
-		}
-		cp = append(cp, jpath)
+	if _, err := os.Stat(jpath); os.IsNotExist(err) {
+		return "", fmt.Errorf("jar %v does not exist", jpath)
 	}
-	return strings.Join(cp, ":"), nil
+	return jpath, nil
 }
 
-func installTemplates(jarsDir string, templateDir string) error {
-	templateZip := path.Join(jarsDir, authTemplates)
-	arch, err := zip.OpenReader(templateZip) // global variable, yech
+func pullTemplatesOutofAuth2Jar(classPath string) (string, error) {
+	dirPath := filepath.Dir(classPath)
+	outfile, err := os.Create(filepath.Join(dirPath, "output.txt"))
 	if err != nil {
-		return err
+		return "", err
+	}
+
+	cmdargs := []string{classPath, "-d", dirPath}
+	cmd := exec.Command("unzip", cmdargs...)
+	cmd.Stdout = outfile
+	cmd.Stderr = outfile
+	err = cmd.Start()
+	if err != nil {
+		return "", err
```

Comment:
There's no way to do this without a system call?

Reply:
If that's the case which seems improbable, surely there's a way to wait until the command exits rather than polling for it to be done?

Reply:
https://stackoverflow.com/a/55043980/643675

Reply:
saw this approach before. Not quite sure how to generate `buff`

Reply:
https://www.tutorialspoint.com/golang-program-to-convert-file-to-byte-array

Reply:
Seems like just opening the file normally without reading it all into memory should work though

Reply:
👍

Reply:
:+1: